[RISC-V] relocation truncated to fit: R_RISCV_JAL against symbol `ThrowExceptionHelper'

### Description

Unable to build runtime with gcc because of compilation error:
```
../../pal/src/libcoreclrpal.a(exceptionhelper.S.o): in function `ThrowExceptionFromContextInternal':
/runtime/src/coreclr/pal/src/arch/riscv64/exceptionhelper.S:47:(.text+0x44): relocation truncated to fit: R_RISCV_JAL against symbol `ThrowExceptionHelper' defined in .text.unlikely.ThrowExceptionHelper section in ../../pal/src/libcoreclrpal.a(seh.cpp.o)
```

### Reproduction Steps

Using Ubuntu 23.10 from Docker (GCC 13.2.0):
```bash
dpkg --add-architecture riscv64 && \
sed -i -E 's|^deb ([^ ]+) (.*)$|deb [arch=amd64] \1 \2\ndeb [arch=riscv64] http://ports.ubuntu.com/ubuntu-ports/ \2|' /etc/apt/sources.list && \
apt-get update && \
apt-get install -y --no-install-recommends \
    ca-certificates \
    python3-minimal \
    python3-libxml2 \
    git \
    wget \
    cmake \
    crossbuild-essential-riscv64 \
    zlib1g-dev:riscv64 \
    libkrb5-dev:riscv64 \
    libssl-dev:riscv64 \
    libicu-dev:riscv64 \
    liblttng-ust-dev:riscv64

ln -s /usr/bin/riscv64-linux-gnu-gcc /usr/bin/gcc
ln -s /usr/bin/riscv64-linux-gnu-g++ /usr/bin/g++
```

```
git clone https://github.com/dotnet/runtime
cd runtime
git checkout v8.0.1
```

```
export ROOTFS_DIR=/
./build.sh -c Release --cross --arch riscv64 --gcc
```

### Expected behavior

Runtime can be compiled with GCC for RISC-V, not only by Clang

### Actual behavior

```
collect2: error: ld returned 1 exit status
```

### Regression?

_No response_

### Known Workarounds

_No response_

### Configuration

_No response_

### Other information

_No response_

Thanks, @am11, that solved the problem. There are link issues later, but initial problem can be workarounded by using `tail`.

```
/usr/lib/gcc-cross/riscv64-linux-gnu/13/../../../../riscv64-linux-gnu/bin/ld: ../jit/static/CMakeFiles/clrjit_obj.dir/__/emitriscv64.cpp.o: in function `GenTree::GetRegNum() const':
/__w/dotnet_riscv/dotnet_riscv/runtime/src/coreclr/jit/gentree.h:928:(.text._ZN7emitter14emitInsTernaryE11instruction8emitAttrP7GenTreeS3_S3_+0x2b6): undefined reference to `CodeGen::genJumpToThrowHlpBlk_la(SpecialCodeKind, instruction, _regNumber_enum, BasicBlock*, _regNumber_enum)'
/usr/lib/gcc-cross/riscv64-linux-gnu/13/../../../../riscv64-linux-gnu/bin/ld: ../jit/static/CMakeFiles/clrjit_obj.dir/__/emitriscv64.cpp.o: in function `CodeGen::rsGetRsvdReg()':
/__w/dotnet_riscv/dotnet_riscv/runtime/src/coreclr/jit/codegen.h:117:(.text._ZN7emitter14emitInsTernaryE11instruction8emitAttrP7GenTreeS3_S3_+0x400): undefined reference to `CodeGen::genJumpToThrowHlpBlk_la(SpecialCodeKind, instruction, _regNumber_enum, BasicBlock*, _regNumber_enum)'
/usr/lib/gcc-cross/riscv64-linux-gnu/13/../../../../riscv64-linux-gnu/bin/ld: ../jit/static/CMakeFiles/clrjit_obj.dir/__/emitriscv64.cpp.o: in function `emitter::emitIns_R_R_I(instruction, emitAttr, _regNumber_enum, _regNumber_enum, long, insOpts)':
/__w/dotnet_riscv/dotnet_riscv/runtime/src/coreclr/jit/emitriscv64.cpp:687:(.text._ZN7emitter14emitInsTernaryE11instruction8emitAttrP7GenTreeS3_S3_+0x6a2): undefined reference to `CodeGen::genJumpToThrowHlpBlk_la(SpecialCodeKind, instruction, _regNumber_enum, BasicBlock*, _regNumber_enum)'
/usr/lib/gcc-cross/riscv64-linux-gnu/13/../../../../riscv64-linux-gnu/bin/ld: /__w/dotnet_riscv/dotnet_riscv/runtime/src/coreclr/jit/emitriscv64.cpp:652:(.text._ZN7emitter14emitInsTernaryE11instruction8emitAttrP7GenTreeS3_S3_+0xc54): undefined reference to `CodeGen::genJumpToThrowHlpBlk_la(SpecialCodeKind, instruction, _regNumber_enum, BasicBlock*, _regNumber_enum)'
collect2: error: ld returned 1 exit status
```

--- a/src/coreclr/jit/codegenriscv64.cpp
+++ b/src/coreclr/jit/codegenriscv64.cpp
@@ -7643,7 +7643,7 @@ void CodeGen::genAllocLclFrame(unsigned frameSize, regNumber initReg, bool* pIni
 #endif
 }
 
-inline void CodeGen::genJumpToThrowHlpBlk_la(
+void CodeGen::genJumpToThrowHlpBlk_la(
     SpecialCodeKind codeKind, instruction ins, regNumber reg1, BasicBlock* failBlk, regNumber reg2)
 {
     assert(INS_beq <= ins && ins <= INS_bgeu);

--- a/src/coreclr/pal/src/arch/riscv64/exceptionhelper.S
+++ b/src/coreclr/pal/src/arch/riscv64/exceptionhelper.S
@@ -44,5 +44,5 @@ LEAF_ENTRY ThrowExceptionFromContextInternal, _TEXT
 
     // The PAL_SEHException pointer
     addi a0, a1, 0
-    j    ThrowExceptionHelper
+    tail ThrowExceptionHelper
 LEAF_END ThrowExceptionFromContextInternal, _TEXT

